### PR TITLE
Add coverage for API function drizzle_free_context

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**

### CODE REVIEW

**Comment by andreas-bok-sociomantic** [comment]
Rebased on latest v6.x.x

File: tests/unit/event_callback.c [hunk 1]
```diff
@@ -81,6 +81,16 @@ extern drizzle_return_t drizzle_event_callback(drizzle_st *con, short events,
   return DRIZZLE_RETURN_OK;
 }
 
+extern void drizzle_free_context(drizzle_st *con, void *context);
+extern void drizzle_free_context(drizzle_st *con, void *context)
+{
+  (void)con;
```
Comment (mathias-lang-sociomantic) [inline tests/unit/event_callback.c:87]: Why do you specify the name then ?

Comment (andreas-bok-sociomantic) [reply tests/unit/event_callback.c:87]: Otherwise it results in an `unused variable warning`

Comment (mathias-lang-sociomantic) [reply tests/unit/event_callback.c:87]: AFAIK compilers won't issue this warning if you don't specify a name:
```
extern void drizzle_free_context(drizzle_st*, void *context) 
```

Comment (andreas-bok-sociomantic) [reply tests/unit/event_callback.c:87]: That results in : 

`../tests/unit/event_callback.c:85:1: error: parameter name omitted`

Comment (andreas-bok-sociomantic) [reply tests/unit/event_callback.c:87]: compiler warning flags are on viagra

Comment (mathias-lang-sociomantic) [reply tests/unit/event_callback.c:87]: That's why I don't want to do C++ anymore

**Pull request merged**